### ISSUE TITLE
Battle2k3: Vampires Dawn II should not play enemy attack sound effect - unknown RPG_RT binary version

### ISSUE THREAD

**Issue by fmatthew5876** [EasyRPG/Player#2476]
When playing "Vampires Dawn II" (RPG Maker 2003 game) with RPG_RT that the enemy_attack_se effect is not played when an enemy attacks. I have checked the LDB database of the game - the enemy_attack_se is set to attack2 with volume and tempo set to 100% and the sound file exists in the Sound folder of the game. You see the enemy flashing once and the attack animation played on the target actor (with its own sound). But the enemy_attack_se effect is not played.

https://www.vampiresdawn.org/vd-2-ancient-blood/

I investigated further and this looks like an RPG_RT.exe binary version difference. If I replace the VD 2 RPG_RT with dynrpg binary, the sound effect plays.

Both the VD 2 RPG_RT.exe, advocate, and dynrpg binaries have FileVersion 1.0.8.0. This game appears to be using advocate, but the hash is different.

```
98149c681fa4f087d770e913a9d58e1f  RPG Maker 2003/Vampires Dawn II - Ancient Blood/RPG_RT.exe
```

```
fdd68cee936ea75c3929b91daa934532  ../../rpgrt/rpg2003/RPG_RT.2003.1.0.3.0.advocate.eng.exe
6b0e33b4e9af49f52e7db66a91174ee3  ../../rpgrt/rpg2003/RPG_RT.2003.1.0.3.0.jp.exe
ea8edc01fea5e776719528c9e6bdaf60  ../../rpgrt/rpg2003/RPG_RT.2003.1.0.8.0.advocate.eng.32bit_color.exe
eed8e0fb848521fcd1699b156e1cc50e  ../../rpgrt/rpg2003/RPG_RT.2003.1.0.8.0.advocate.eng.dynrpg.exe
ff0fb3b085047f93646832a762d2a883  ../../rpgrt/rpg2003/RPG_RT.2003.1.0.8.0.advocate.eng.exe
c87407cb68c51e87e2e19c3e5ac0b416  ../../rpgrt/rpg2003/RPG_RT.2003.1.0.9.1.fdelapena.spa.cleartype.exe
fa15d283766197c0e8833c48b69a60fc  ../../rpgrt/rpg2003/RPG_RT.2003.1.0.9.1.unknown.eng.exe
ad0e8f0dd16ca6cf88de1db55b78d840  ../../rpgrt/rpg2003/RPG_RT.2003.1.1.2.0.eng.exe
9373d938e1b79db88a6ddac9640bee98  ../../rpgrt/rpg2003/RPG_RT.2003.1.1.2.1.eng.exe
```

I took a peak at the VD 2 binary in RE tools and it is indeed different, very different at least in the logic that triggers the enemy SE effect.

In DynRPG the enemy flash and attack sound effect happens unconditionally in:

```
ProcessBattleAction (0049A04C)
SetupMonsterAction (0045D7D1)
GetEnemyAttackSE (0048BC84)
```

This is also the same for 2k3e

In the VD 2 binary:
*  `ProcessBattleAction (0049A04C)` just does `SetFlash (9004C1178)` inline
* The call to `GetEnemyAttackSE 0048BC84` happens in `RPG_Monsters::UpdateBattle (004BEE88)` during animation processing

This behavior also occurs in the 1.0.3.0 version of RPG_RT.exe. I checked the one I have an its Resource FileVersion says 1.0.4.0

So in summary, VD 2 `RPG_RT.exe` appears to be the same version as advocate, but at least has some legacy RPG_RT.exe behavior w.r.t. the battle system.

So in order to fix this in Player we need to:
- [ ] Identify exactly which RPG_RT version is used for VD 2
- [ ] Make Player detect it properly (currently detects it as MajorUpdated)
- [ ] Modify battle system code to not play the enemy SE effect for this unknown version

**Comment by fmatthew5876** [comment]
I've uploaded the binary here: https://github.com/fmatthew5876/rpgrt/blob/master/rpg2003/RPG_RT.2003.1.0.8.0.vampires.dawn.II.unknown.eng.exe

**Comment by CherryDT** [comment]
It was an RM2k3 bug that the sound effect before was never played at all, so it's possible that there are games with a sound set that should actually not be played.

As you noticed, I fixed this in DynRPG and in v1.11.

If there are things like these were games rely on certain bugs to be present, maybe there needs to be tighter emulation of the different versions.

You can find the changelogs here (both mention this change among several others):

* DynRPG patch changelog: http://rpg-maker.cherrytree.at/dynrpg/patch.html
* RM2k3E changelog: https://docs.google.com/document/d/1sXlXZIXdjxVq5gh_lPFuFLMbMRzZe4BKPpRI13ZD2Qc/edit#heading=h.love8s87o4rc

**Comment by CherryDT** [comment]
It was an RM2k3 bug that the sound effect before was never played at all, so it's possible that there are games with a sound set that should actually not be played.

As you noticed, I fixed this in DynRPG and in v1.11. So, any non-DynRPG RPG_RT.exe with a version of less than 1.11 (up to 1.10a) should have this bug (unless it was patched with a separate patch, not sure if I ever released one) **but** it seems you discovered that older versions actually didn't have this bug! That's interesting, I didn't know that. Maybe it was introduced in 1.05 where a ton of changes were done... 1.08-1.10a definitely have the bug though, not sure at which point it was originally introduced then.

If there are things like these were games rely on certain bugs to be present, maybe there needs to be tighter emulation of the different versions.

You can find the changelogs here (both mention this change among several others):

* Changes done to the engine by DynRPG patch: http://rpg-maker.cherrytree.at/dynrpg/patch.html
* RM2k3E changelog: https://docs.google.com/document/d/1sXlXZIXdjxVq5gh_lPFuFLMbMRzZe4BKPpRI13ZD2Qc/edit

**Comment by fmatthew5876** [comment]
Ahh that solves the mystery then.

I've been looking at DynRPG assuming it's the same as 1.0.8.0. I didn't know your DynRPG patch also included bug fixes for RPG_RT. A native advocate binary indeed has this bug. I'll update my PR accordingly.
